core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -10769,6 +10769,12 @@ private void checkCustomColumnResolving(String table) {
         "(?s).*Expected a character type*");
   }
 
+  @Test public void testJsonType() {
+    check("select json_type(ename) from emp");
+    checkExp("json_type('{\"foo\":\"bar\"}')");
+    checkExpType("json_type('{\"foo\":\"bar\"}')", "VARCHAR(20) NOT NULL");
```

Comment:
I prefer adding a case for something like `json_type(1)`, which should fail the validation.